### PR TITLE
PR: Fix unwanted scrolling when selecting tabs

### LINKED ISSUE #4597: Editors not gaining focus when changing by tab click
With 2 or more files open on the editor, clicking on different tabs should give the focus to the editor (cursor should start blinking) and the outline explorer should update because of this focus gain. Currently the focus is not given to the editor. This does works with the file switcher.

This is on 3.x but not on 3.1.x.

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes #4597 
Original bug was #1170

The solution implemented in 2228e83b656a476683f27e9ddd5e234a08135be6 wasn't the best, focus should be give to the Editor when changing tabs.

I reverted 2228e83b656a476683f27e9ddd5e234a08135be6, and traced the error until `self.tabs.setTabText(index, tab_text)` I'm not sure why this line was causing that error, maybe it raises again the `currentChanged` signal.

### CODE REVIEW

**Comment by goanpeca** [comment]
Will test later today thanks @rlaverde

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -1210,7 +1210,11 @@ def set_stack_title(self, index, is_modified):
         is_readonly = finfo.editor.isReadOnly()
         tab_text = self.get_tab_text(index, is_modified, is_readonly)
         tab_tip = self.get_tab_tip(fname, is_modified, is_readonly)
-        self.tabs.setTabText(index, tab_text)
+
+        # Only update tab text if have changed, otherwise an unwanted scrolling
+        # will happen when changing tabs, see issue: 1170
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Please change this line to

    # will happen when changing tabs. See Issue #1170.

**Comment by ccordoba12** [comment]
The failure in Appveyor was unrelated, so I restarted it.

**Comment by ccordoba12** [comment]
@goanpeca, please test and approve this one.

**Pull request merged**